### DISCUSSION TITLE
How to retrieve trips data out of PSA Controller or in general which APIs are exposed to HA by psa_car_controller

### DISCUSSION THREAD

**Discussion by malbertone1** [flobz/psa_car_controller#835] [Q&A]
Hi,
I have installed the Add On and it's working fine to retrieve data using:
curl -v http://0.0.0.0:5000/get_vehicleinfo/<My VIN>

but can see that no information is available about the stored trips.

When looking at the PSA Controller dashboard i have the trip list so guess there should be a way to retrieve them when creating the sensors or other components relying on the data provided by the psa_car_controller.

My question could be more in general about the API specs exposed by PSA Car Controller to retrieve the data

thanks a lot!
Mike

**Comment by sod75** [comment]
Check the ReadMe,
https://github.com/flobz/psa_car_controller?tab=readme-ov-file
Section II:
https://github.com/flobz/psa_car_controller/blob/master/docs/psacc_api.md
17.  Get the vehicle trips:
http://localhost:5000/vehicles/trips

It's a recent addition though, make sure you're running the latest version

**Reply by malbertone** [reply]
Thanks a lot, appreciated, yes running the latest version.
Cheers,

Mike
